Source/Transform/KTFractionalFFT.cc

```diff
@@ -0,0 +1,278 @@
+
+        if (! fForwardFFT.InitializeForComplexTDD())
+        {
+            KTERROR(evlog, "Error while initializing the forward FFT");
+            return false;
+        }
+
+        if (! fReverseFFT.InitializeForComplexTDD())
+        {
+            KTERROR(evlog, "Error while initializing the reverse FFT");
+            return false;
+        }
+
+        fInitialized = true;
+        return true;
+    }
+
+    bool KTFractionalFFT::ProcessTimeSeriesChirpOnly( KTTimeSeriesData& tsData, KTSliceHeader& slice )
+    {
+        KTDEBUG(evlog, "Receiving time series for fractional FFT");
+
+        KTTimeSeriesFFTW* ts = nullptr;         // time series from data
+        
+        double norm = 0.;                       // norm of the current TS value
+        double phase = 0.;                      // argument of current TS value
+        double binOffset = slice.GetTimeInAcq() / (double)slice.GetBinWidth();
+        double q = fSlope * KTMath::Pi() * slice.GetBinWidth() * slice.GetBinWidth();
+
+        for( unsigned iComponent = 0; iComponent < tsData.GetNComponents(); ++iComponent )
+        {
+            KTDEBUG(evlog, "Processing component: " << iComponent);
+
+            // get TS from data object and make the new TS
+            ts = dynamic_cast< KTTimeSeriesFFTW* >(tsData.GetTimeSeries( iComponent ));
+            if( ts == nullptr )
+            {
+                KTWARN(evlog, "Couldn't find time series object. Continuing to next component");
+                continue;
+            }
+            
+            // Loop through all bins
+            for( unsigned iBin = 0; iBin < ts->GetNTimeBins(); ++iBin )
+            {
+                // Obtain norm and phase from components
+                norm = std::hypot( (*ts)(iBin)[0], (*ts)(iBin)[1] );
+                phase = atan2( (*ts)(iBin)[1], (*ts)(iBin)[0] );
+
+                // Shift phase
+                phase -= q * ((double)iBin + binOffset) * ((double)iBin + binOffset);
+
+                // Assign components from norm and new phase
+                (*ts)(iBin)[0] = norm * cos( phase );
+                (*ts)(iBin)[1] = norm * sin( phase );
+            }
+        }
+
+        return true;
+    }
+
+    bool KTFractionalFFT::ProcessTimeSeries( KTTimeSeriesData& tsData, KTTimeSeriesData& newTSData, KTFrequencySpectrumDataFFTW& newFSData, KTSliceHeader& slice )
+    {
+        KTDEBUG(evlog, "Receiving time series for fractional FFT");
+
+        KTTimeSeriesFFTW* ts = nullptr;   // time series from data
+        KTTimeSeriesFFTW* tsDup = new KTTimeSeriesFFTW( slice.GetSliceSize(), 0.0, slice.GetSliceLength() );
+        KTTimeSeriesFFTW* newTS;
+        KTFrequencySpectrumFFTW* fs;
+        KTFrequencySpectrumFFTW* newFS = new KTFrequencySpectrumFFTW( slice.GetSliceSize(), 0.0, slice.GetSampleRate() );
+
+        newFS->SetNTimeBins( slice.GetSliceSize() );
+
+        KTINFO(evlog, "q1 N/pi = " << tan( 0.5 * fAlpha ));
```

Comment:
Should these really be KTINFO?

Unless you're going to explain what the variable `q` is (e.g. with more explanation in the header), please choose a longer name for `q1` and `q2`.

Reply:
I think they are extraneous anyway. fAlpha is printed right after, to KTINFO, when it is calculated if necessary. I'll leave that in and remove these.